src/main/java/com/example/MyBatisDemo/repository/StudentRepository.java

```diff
@@ -0,0 +1,19 @@
+package com.example.MyBatisDemo.repository;
+
+import com.example.MyBatisDemo.model.StudentEntity;
+import org.apache.ibatis.annotations.Mapper;
+import org.apache.ibatis.annotations.Param;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+@Mapper
+@Transactional
```

Comment:
Is it right to add transactional annotation on the repo?
Where should the transactional annotation be ideally?

Reply:
Ideally it should be over the Services
I forgot to remove it from repositories. 
Although we remove transactional annotation, application runs smoothly without error